schema/resource.go

```diff
@@ -91,6 +95,41 @@ func (r *Resource) Columns() []string {
 	return r.Table.Columns.Names()
 }
 
+//nolint:revive
+func (r *Resource) CalculateCQID(deterministicCQID bool) error {
+	if !deterministicCQID {
+		uuidGen := uuid.New()
+		b, err := uuidGen.MarshalBinary()
+		if err != nil {
+			return err
+		}
+		return r.Set(CqIDColumn.Name, b)
```

Comment:
[nit]
We do the same `MarshalBinary` -> check `err` -> `return r.Set(CqIDColumn.Name, b)` at the end of the func.
Maybe we could extract the uuid func to return either `uuid.New()` or deterministic SHA1-based one and then just call `MarshalBinary` on the returned value?